tests/foreman/ui_airgun/test_discoveredhost.py

```diff
@@ -0,0 +1,496 @@
+# -*- encoding: utf-8 -*-
+"""Test class for Foreman Discovery
+
+:Requirement: Discoveredhost
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: UI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from fauxfactory import gen_string
+from nailgun import entities
+
+from robottelo import ssh
+from robottelo.api.utils import configure_provisioning
+from robottelo.decorators import (
+    fixture,
+    run_in_one_thread,
+    run_only_on,
+    skip_if_not_set,
+    tier3,
+    upgrade,
+)
+from robottelo.libvirt_discovery import LibvirtGuest
+from robottelo.products import RHELRepository
+
+
+@fixture(scope='module')
+def module_org():
+    return entities.Organization().create()
+
+
+@fixture(scope='module')
+def module_loc(module_org):
+    return entities.Location(
+        name=gen_string('alpha'),
+        organization=[module_org],
+    ).create()
+
+
+@fixture(scope='module')
+def provisioning_env(module_org, module_loc):
+    # Build PXE default template to get default PXE file
+    entities.ConfigTemplate().build_pxe_default()
+    # Update default org and location params to place discovered host
+    discovery_loc = entities.Setting().search(
+        query={'search': 'name="discovery_location"'})[0]
+    default_discovery_loc = discovery_loc.value
+    discovery_loc.value = module_loc.name
+    discovery_loc.update(['value'])
+    discovery_org = entities.Setting().search(
+        query={'search': 'name="discovery_organization"'})[0]
+    default_discovery_org = discovery_org.value
+    discovery_org.value = module_org.name
+    discovery_org.update(['value'])
+    # Enable flag to auto provision discovered hosts via discovery rules
+    discovery_auto = entities.Setting().search(
+        query={'search': 'name="discovery_auto"'})[0]
+    default_discovery_auto = str(discovery_auto.value)
+    discovery_auto.value = 'True'
+    discovery_auto.update(['value'])
+    # we need the default RHEL os version
```

Comment:
i don't think so - we don't use this at all. We do not do a proper provisioning anywhere in the below tests. So we can make some custom os, media and everything as we do it for `make_host()`

Reply:
@rplevka in the factory provisioning setup used here rely on creating a repo and choosing an os.

Reply:
@rplevka and if the os is wrongly chosen the factory provisioning setup will fail.

Reply:
@ldjebran - my point was - you don't need to configure any provisioning at all, just don't call the factory.

Reply:
@rplevka I'am a bit confused here, as the goal of this module is to discover real hosts.

Reply:
@ldjebran sorry for my last comment. Why don't you simply create some custom dummy repo + medium and use it?

you saide the following:
>  the factory provisioning setup used here rely on creating a repo and choosing an os.

but i'm confused - what factory are you talking about? can you point me to it?
i'm trying to say that syncing the real RH repo is a waste since we don't do any real provisioning anywhere => there's no real VM that would really need the packages.
All we test is that foreman creates the Host entry.

Reply:
As understood, the cli discovery tests are using medium and api tests are using repo + content view  that enable to use kickstart_repository (this scenario is not working with a dummy repo)
I'am receiving an error 
```
"error": {"id":null,"errors":{"name":[],"base":["The selected kickstart repository is not part of the assigned content view, lifecycle environment,\n                  content source, operating system, and architecture"]},"full_messages":["The selected kickstart repository is not part of the assigned content view, lifecycle environment,\n                  content source, operating system, and architecture"]}
```
CLI factory : https://github.com/SatelliteQE/robottelo/blob/master/robottelo/cli/factory.py#L3312
API factory: https://github.com/SatelliteQE/robottelo/blob/master/robottelo/api/utils.py#L298

Perhaps this was intentional by design to test the two scenarios.

Reply:
@ldjebran,
ok Djebran, thanks for the explanation and sorry for holding this for long. i guess, let's proceed and merge the commit.
If I found ways of improving the tests, I'll submit a PR.